### DISCUSSION TITLE
posts/grafana-loki-kubernetes/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#271] [Comments]
# posts/grafana-loki-kubernetes/

In my previous video (Meet Grafana LOKI, a log aggregation system for everything and post, I promised that I would also explain how to install Granfana Loki on Kubernetes using helm.If you’re looking to set this up in docker-compose, be sure to check out this video

https://technotim.live/posts/grafana-loki-kubernetes/

**Comment by cocuba** [comment]
thank you.  I had installed the stack but without persistence, with this post i enabled it.

**Comment by jfmatth** [comment]
Love the video and this post.  Any reason why syslog isn't working for me?

**Comment by ASMACOSNMX** [comment]
I followed it to the letter, no luck accesing http from other servers / networks, no firewall in place for testing purposes.

Any ideas? Thanks for such AWESOME guide, btw.

**Comment by ghost** [comment]
logger=tsdb.loki endpoint=checkHealth pluginId=loki dsName=Loki dsUID=b44f164b-ba45-46db-80d7-d2139ccc079b uname=admin fromAlert=false t=2023-10-31T14:59:05.401157496Z level=error msg="Error received from Loki" duration=1.848241ms stage=databaseRequest statusCode=400 contentLength=65 start=1970-01-01T00:00:01Z end=1970-01-01T00:00:04Z step=1s query=vector(1)+vector(1) queryType=instant direction=backward maxLines=0 supportingQueryType=none lokiHost=loki:3100 lokiPath=/loki/api/v1/query status=error error="parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"
logger=tsdb.loki endpoint=checkHealth pluginId=loki dsName=Loki dsUID=b44f164b-ba45-46db-80d7-d2139ccc079b uname=admin fromAlert=false t=2023-10-31T14:59:05.4012166Z level=error msg="Error querying loki" error="parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"
logger=tsdb.loki endpoint=CheckHealth t=2023-10-31T14:59:05.401264428Z level=error msg="Loki health check failed" error="error from loki: parse error at line 1, col 1: syntax error: unexpected IDENTIFIER"
logger=context userId=1 orgId=1 uname=admin t=2023-10-31T14:59:05.401378167Z level=info msg="Request Completed" method=GET path=/api/datasources/uid/b44f164b-ba45-46db-80d7-d2139ccc079b/health status=400 remote_addr=10.25.1.4 time_ms=5 duration=5.756894ms size=106 referer=http://10.25.1.131:3000/connections/datasources/edit/b44f164b-ba45-46db-80d7-d2139ccc079b handler=/api/datasources/uid/:uid/health



its throwing this error with exact same config you provided above.

**Reply by ghost** [reply]
unable to add loki data source from grafana.

**Comment by sojjan1337** [comment]
How do i include local logs in this setup. I can only see the logs for the pods.
Thanks!

**Comment by firatoezcan** [comment]
Hey, just wanted to mention that loki-stack is deprecated. Obviously if it still works, no harm in continue using it, however after me spending like, 3 days on it, I think it could make for a good candidate to do a followup video.

Your content is still amazing though and the Prometheus video helped me out a lot, thanks for doing what you do!!

**Reply by madejackson** [reply]
Thanks, your comment led me on the right path. After installing my k3s mainly with TT's guides I was able to install loki with the following cmd on my k3s + Longhorn + Prometheus:

helm upgrade --install loki grafana/loki -n monitoring --set singleBinary.persistence.enabled=true   --set singleBinary.persistence.storageClass=longhorn   --set singleBinary.persistence.size=20Gi --set loki.auth_enabled=false --set singleBinary.replicas=3

**Comment by Sneevus** [comment]
If you want to use the latest version of loki (3.2.1 at time of posting), you need to change the `schema_config` section in `loki-config.yml` to update the **store** and **schema** values, or the docker container will just be stuck in an endless restart cycle.

The following example fragment worked for me.

```yaml
schema_config:
  configs:
    - from: 2024-11-13
      store: tsdb
      object_store: filesystem
      schema: v13
      index:
        prefix: index_
        period: 24h
```
